theano/tensor/sort.py

```diff
@@ -62,39 +62,31 @@ def grad(self, inputs, output_grads):
             self, 0, axis,
             "Currently, we only implement the gradient on sort for vector"
             " matrix (and axis is None or 0) and tensor3")
-        if a.ndim == 1:
-            idx = argsort(*inputs, kind=self.kind, order=self.order)
-            # rev_idx = numpy.where(idx[None, :]==numpy.arange(5)[:,None])[1]
-            rev_idx = theano.tensor.eq(idx[None, :],
-                                       arange(a.shape[0])[:, None]).nonzero()[1]
-            inp_grad = output_grads[0][rev_idx]
-        elif a.ndim == 2:
-            if (axis is None or
-                    (isinstance(axis, theano.Constant) and axis.data is None)):
-                idx = argsort(*inputs, kind=self.kind, order=self.order)
-                rev_idx = theano.tensor.eq(
-                    idx[None, :],
-                    arange(a.shape[0] * a.shape[1])[:, None]).nonzero()[1]
-                inp_grad = output_grads[0][rev_idx].reshape(a.shape)
-            elif (axis == 0 or
-                  (isinstance(axis, theano.Constant) and axis.data == 0)):
-                idx = argsort(*inputs, kind=self.kind, order=self.order)
-                # not working: numpy.where(idx[None, :]==numpy.arange(2)[:, None, None])
-                pass
-        elif a.ndim == 3:
-            if isinstance(axis, theano.Constant) and axis.data is not None:
-                indices = self.__get_argsort_indices(a, axis)
-                inp_grad = output_grads[0][indices[0], indices[1], indices[2]]
-            elif (axis is None or
-                    (isinstance(axis, theano.Constant) and axis.data is None)):
-                rev_idx = self.__get_argsort_indices(a, axis)
-                inp_grad = output_grads[0][rev_idx].reshape(a.shape)
+        if ((isinstance(axis, theano.Constant) or (isinstance(axis,
+                                                   theano.tensor.
+                                                   TensorVariable) and
+                                                   axis.ndim == 0)) and
+                axis.data is not None):
+            indices = self.__get_argsort_indices(a, axis)
+            inp_grad = output_grads[0][tuple(indices)]
```

Comment:
Do you think you could call `tuple(...)` before returning the value from `__get_argsort_indices` when axis is not None? That way, these two lines could be combined with the `elif` block.